src/stk/_internal/topology_graphs/polymer/vertices.py

```diff
@@ -204,3 +209,110 @@ def place_building_block(self, building_block, edges):
 
     def map_functional_groups_to_edges(self, building_block, edges):
         return {fg_id: edge.get_id() for fg_id, edge in enumerate(edges)}
+
+
+class CoreVertex(Vertex):
+    """
+    Represents a vertex in the core of an ncore topology graph.
+
+    """
+
+    def place_building_block(self, building_block, edges):
```

Comment:
add type annotations

Reply:
I did this, but consider that the base class has type hints, so it is not necessary. However, some extra changes were. Please re review.